### ISSUE TITLE
Pushy Notification Threads get Stuck for future.get

### ISSUE THREAD

**Issue by vsharm22** [jchambers/pushy#815]
Hello ,

Too many notification jvm Thread hungs and Application down in every 24 hours . Pushy version used 0.13.9.

Below is the Application code base :

  Future<PushNotificationResponse<ApnsNotificationMessage>> sendNotificationFuture = clientToSendNotification.sendNotification(apnsMessage);
            
 pushNotificationResponse = sendNotificationFuture.get();

We took the thread dump and below is thread dump logs:

"pool-3-com.company.task.TaskExecutor$WorkerGroup-thread-2" Id=117 WAITING on com.turo.pushy.apns.PushNotificationPromise@5c2af966
	at java.lang.Object.wait(Native Method)
	-  waiting on com.turo.pushy.apns.PushNotificationPromise@5c2af966
	at java.lang.Object.wait(Object.java:502)
	at io.netty.util.concurrent.DefaultPromise.await(DefaultPromise.java:218)
	at io.netty.util.concurrent.DefaultPromise.await(DefaultPromise.java:32)
	at io.netty.util.concurrent.AbstractFuture.get(AbstractFuture.java:32)
	at com.company.task.dispatcher.notification.ApplePushNotificationHTTP2Channel.sendNotification(ApplePushNotificationHTTP2Channel.java:605)
	at com.company.task.dispatcher.notification.ChannelWrapper.sendNotification(ChannelWrapper.java:97)
	at com.company.task.dispatcher.notification.NotificationChannelManagerImpl.doSendNotification(NotificationChannelManagerImpl.java:201)

**Comment by jchambers** [comment]
Hello! This is a duplicate of #815. We've consolidated discussion around the issue into #816. Let's continue the conversation there.
